tests/integration/test_fim/test_files/test_follow_symbolic_link/test_audit_rules_removed_after_change_link.py

```diff
@@ -0,0 +1,106 @@
+
+from test_fim.test_files.test_follow_symbolic_link.common import configurations_path, testdir1, \
+    modify_symlink, testdir_not_target, test_directories, wait_for_audit
+
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=1)]
+
+wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+# Variables
+
+fname = "testfile"
+symlink_root_path = "/"
+symlink_name = "symlink"
+symlink_path = os.path.join(symlink_root_path, symlink_name)
+link_interval = 2
+
+param_dir = {
+    'FOLLOW_MODE': 'yes',
+    'LINK_PATH': symlink_path
+}
+
+# Configurations
+
+conf_params, conf_metadata = generate_params(extra_params=param_dir, modes=['whodata'])
+configurations = load_wazuh_configurations(configurations_path, __name__, params=conf_params, metadata=conf_metadata)
+
+# Functions
+
+
+def extra_configuration_before_yield():
+    """
+    Setup the symlink to one folder
+    """
+    # Symlink pointing to testdir1
+    create_file(SYMLINK, symlink_root_path, symlink_name, target=testdir1)
+    # Set symlink_scan_interval to a given value
+    change_internal_options(param='syscheck.symlink_scan_interval', value=link_interval)
+
+
+def extra_configuration_after_yield():
+    """
+    Setup the symlink to one folder
+    """
+    # Symlink pointing to testdir1
+    os.remove(symlink_path)
+    change_internal_options(param='syscheck.symlink_scan_interval', value=600)
+
+
+# fixtures
+
+@pytest.fixture(scope='module', params=configurations)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+# tests
+
+@pytest.mark.parametrize('replaced_target, new_target, file_name, tags_to_apply', [
+                         (testdir1, testdir_not_target, f'{fname}_1', {'check_audit_removed_rules'})
+                         ])
+def test_audit_rules_removed_after_change_link(replaced_target, new_target, file_name, tags_to_apply,
+                                               get_configuration, configure_environment,
+                                               restart_syscheckd, wait_for_fim_start):
+    """
+    Parameters
+    ----------
+    main_folder : str
+        Directory that is being pointed at or contains the pointed file.
+    aux_folder : str
+        Directory that will be pointed at or will contain the future pointed file.
+    """
+    check_apply_test(tags_to_apply, get_configuration['tags'])
+    create_file(REGULAR, replaced_target, file_name)
+    ev = wazuh_log_monitor.start(timeout=global_parameters.default_timeout, callback=callback_detect_event,
+                                 error_message='Did not receive expected "Sending FIM event: ..." event').result()
+
+    assert ev['data']['type'] == 'added' and ev['data']['path'] == os.path.join(replaced_target, file_name)
+
+    # Change the target of the symlink and expect events while there's no syscheck scan
+
+    modify_symlink(new_target, symlink_path)
+
+    wait_for_audit(True, wazuh_log_monitor)
```

Comment:
We are missing a call to `wait_for_symlink_check` prior to this line to guarantee that the symbolic link is updated before the audit rules are reloaded.